### DISCUSSION TITLE
Refused to create a worker because it violates the following Content Security Policy directive: worker-src

### DISCUSSION THREAD

**Discussion by cmacdonnacha** [mswjs/msw#1715] [Q&A] [answered]
Keep seeing this error in the console. MSW still runs, but would like to find the source of the issue.

```
Refused to create a worker from 'http://localhost:3000/mockServiceWorker.js' because it violates the following Content Security Policy directive: "worker-src blob:"."
```

![image](https://github.com/mswjs/msw/assets/9777763/994d952a-52df-44a5-abb4-c2da35a6a118)

**Comment by kettanaito** [comment, marked as answer]
Hi, @cmacdonnacha. 

## Why is this happening?

It looks like your server has a strict `Content-Security-Policy` defined that only allows certain origins for `worker-src`. To my best knowledge, `worker-src blob:` translates to: only allow installing workers at the `blob:` identifier. Since MSW worker is installed using the `http:` scheme, it violates the set security policy, raising the error you see. 

## How to resolve it?

First, see if you can adjust the `Content-Security-Policy` values on your server. Those make sense in production but you may have a looser set for development to allow tools like MSW. 

If you don't own the server or cannot adjust the said policy, you can look into browser-level whitelisting rules for that. There's nothing we can do on MSW side since it isn't the library's issue.

**Reply by cmacdonnacha** [reply]
Thanks @kettanaito
